### PR TITLE
Add caller info to 33across bid requests.

### PR DESCRIPTION
**Pull request by CTMBNara**

### CODE REVIEW

File: src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java [hunk 1]
```diff
@@ -39,16 +42,19 @@
...
 
-public class TtxBidder implements Bidder<BidRequest> {
+public class ThirtyThreeAcrossBidder implements Bidder<BidRequest> {
 
-    private static final TypeReference<ExtPrebid<?, ExtImpTtx>> TTX_EXT_TYPE_REFERENCE =
+    private static final TypeReference<ExtPrebid<?, ExtImpThirtyThreeAcross>> THIRTY_THREE_ACROSS_EXT_TYPE_REFERENCE =
             new TypeReference<>() {
             };
+    private static final ThirtyThreeAcrossExtTtxCaller PREBID_CALLER =
+            ThirtyThreeAcrossExtTtxCaller.of("Prebid-Server", "n/a");
```
Comment (curlyblueeagle) [inline src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java:51]: Hi, I am the maintainer from 33Across. Can I request the caller name passed as `Prebid-Server-Java` to differentiate requests from the Go adapter we already have? Thanks.

**Review by yevhenii-viktorov — APPROVED** [review]

File: src/main/java/org/prebid/server/bidder/thirtythreeacross/proto/ThirtyThreeAcrossExtTtxCaller.java [hunk 2]
```diff
@@ -0,0 +1,14 @@
+package org.prebid.server.bidder.thirtythreeacross.proto;
+
+import com.fasterxml.jackson.annotation.JsonInclude;
+import lombok.Value;
+
+@Value(staticConstructor = "of")
+public class ThirtyThreeAcrossExtTtxCaller {
+
+    @JsonInclude(JsonInclude.Include.NON_EMPTY)
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/thirtythreeacross/proto/ThirtyThreeAcrossExtTtxCaller.java:9]: Pls remove @JsonInclude(JsonInclude.Include.NON_EMPTY)

File: src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java [hunk 3]
```diff
@@ -59,52 +65,88 @@ public TtxBidder(String endpointUrl, JacksonMapper mapper) {
...
+        return modifiedExt;
+    }
+
+    private static ThirtyThreeAcrossExtTtx modifyRequestExtTtx(ThirtyThreeAcrossExtTtx extTtx) {
+        final List<ThirtyThreeAcrossExtTtxCaller> callers;
+        if (extTtx == null || CollectionUtils.isEmpty(extTtx.getCaller())) {
+            callers = Collections.singletonList(PREBID_CALLER);
+        } else {
+            callers = new ArrayList<>(extTtx.getCaller());
+            callers.add(PREBID_CALLER);
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java:121]: ListUtils.union(extTtx.getCaller(), Collections.singletonList(PREBID_CALLER)) and then create ternary

File: src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java [hunk 4]
```diff
@@ -59,52 +65,88 @@ public TtxBidder(String endpointUrl, JacksonMapper mapper) {
...
+        try {
+            return request.toBuilder().ext(modifyRequestExt(request.getExt())).build();
+        } catch (PreBidException e) {
+            errors.add(BidderError.badInput(e.getMessage()));
+            return request;
+        }
+    }
+
+    private ExtRequest modifyRequestExt(ExtRequest ext) throws PreBidException {
+        final Map<String, JsonNode> properties = ObjectUtil.getIfNotNull(ext, ExtRequest::getProperties);
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java:97]: extProperties

File: src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java [hunk 5]
```diff
@@ -59,52 +65,88 @@ public TtxBidder(String endpointUrl, JacksonMapper mapper) {
...
-            return mapper.mapper().convertValue(imp.getExt(), TTX_EXT_TYPE_REFERENCE).getBidder();
+            return mapper.mapper().convertValue(imp.getExt(), THIRTY_THREE_ACROSS_EXT_TYPE_REFERENCE).getBidder();
         } catch (IllegalArgumentException e) {
             throw new PreBidException(e.getMessage());
         }
     }
 
-    private Imp updateImp(Imp imp, ExtImpTtx extImpTtx) throws PreBidException {
-        final String productId = extImpTtx.getProductId();
+    private Imp updateImp(Imp imp, ExtImpThirtyThreeAcross extImpThirtyThreeAcross) throws PreBidException {
```
Comment (SerhiiNahornyi) [inline src/main/java/org/prebid/server/bidder/thirtythreeacross/ThirtyThreeAcrossBidder.java:142]: Pls remove unchecked exceptions from method declarations

**Review by SerhiiNahornyi — CHANGES_REQUESTED** [review]

**Review by yevhenii-viktorov — APPROVED** [review]

**Pull request merged**